src/request-pipeline/index.js

```diff
@@ -96,7 +96,11 @@ const stages = {
                         ctx.onResponseEventDataWithoutBody.push({ rule, opts: configureResponseEvent.opts });
                 });
 
-                ctx.destRes.pipe(ctx.res);
+                if (ctx.contentInfo.isNotModified)
+                    ctx.res.end('');
```

Comment:
Can we simplify this to `ctx.res.end('');`

I guess we need do same for requests with processed content too.